### ISSUE TITLE
Setting up an outliner blog on a personal domain (narrated progress & questions)

### ISSUE THREAD

**Issue by brandon-toner** [scripting/Scripting-News#210]
Currently working on getting setup with an outliner-style blog. Using this thread to post milestones and questions.

**Comment by brandon-toner** [comment]
Next steps: 
• Create a server on Glitch
• Map my personal domain to that server

**Comment by brandon-toner** [comment]
Not sure which of these I need to choose...
My best guess is node.js minimal. (But I'm not sure what the difference is between minimal and "hello node")

Gonna open both and see what the differences are.
![image](https://user-images.githubusercontent.com/67168828/122683661-0bbc5d80-d1d7-11eb-82cb-0178c10685f3.png)

**Comment by scripting** [comment]
Brandon you definitely want node. Can you install it without either of the two of what are they called starter apps? You just want plain old node.

**Comment by brandon-toner** [comment]
<pardon the traffic here, I'm going to document much of my thinking and steps>

1) Went to https://www.cira.ca/ thinking that was my domain registry (.ca), but realized it's probably namecheap.com.
2) Went to namecheap.com, and "redirected" brandontoner.ca to my current page location (for an interim measure)

Next: Play around with Glitch then figure out how to:
1) Upload PagePark to that server
2) Map my domain to that server

![image](https://user-images.githubusercontent.com/67168828/122686724-1b44a200-d1e9-11eb-8218-06117f64d56d.png)

**Comment by brandon-toner** [comment]
I suspect this (image) is the place where I'll need to map the brandontoner.ca domain to the server domain. 
Not sure the syntax of what might go in "nameserver 1/2" but I'mm figure that out with some googling. 
![image](https://user-images.githubusercontent.com/67168828/122686771-6494f180-d1e9-11eb-952a-7ee8324e9f36.png)

**Comment by brandon-toner** [comment]
Ahh! This looks promising.

![image](https://user-images.githubusercontent.com/67168828/122687009-8e024d00-d1ea-11eb-8f04-97eb32f24220.png)

**Comment by brandon-toner** [comment]
Ok, mapped using a CNAME, I think.
![image](https://user-images.githubusercontent.com/67168828/122687144-4f20c700-d1eb-11eb-9c4c-03555e79668d.png)

**Comment by brandon-toner** [comment]
Ok that worked!! (The default program when I created the server loaded when calling my domain)

![image](https://user-images.githubusercontent.com/67168828/122687313-2220e400-d1ec-11eb-9df4-f5776c4758e7.png)

**Comment by brandon-toner** [comment]
Ok, now I need to load PagePark onto the server — that part I don't know how to do.

**Comment by scripting** [comment]
Brandon, I don’t think you load pagepark, I think you point glitch at the GitHub repo for pagepark.

**Comment by brandon-toner** [comment]
Ohhhh, that make sense. How do I do that?

**Comment by scripting** [comment]
I have a little time now to look into this.

https://github.com/scripting/Scripting-News/issues/168

That's a thread where I was asking for (and getting) help with glitch, from
april 2020.

**Comment by scripting** [comment]
Brandon, here's a link to my running PagePark server.

http://pagepark.glitch.me/

**Comment by scripting** [comment]
Brandon, I think this link will let you edit my project, please just look, and I think there will also be a button where you can "remix" it, that is copy my project and make it your own.

https://glitch.com/edit/#!/pagepark

If you get it running you should have a running PagePark.

Then you can do the thing with the CNAME, and create a sub-folder of the domains folder with the name of your domain, put an OPML file in there, and you're on the air.

Then, later -- tomorrow probably, we'll set up a config.json file in your domain folder that will tell PP where to find the actual OPML file you want to server, that will be on the littleoutliner.com domain.

**Comment by scripting** [comment]
I added a link to this thread from scripting.com.

http://scripting.com/2021/06/20.html#a212838

**Comment by brandon-toner** [comment]
Success! (I have the Little Outliner OPML file in my domain folder)

![image](https://user-images.githubusercontent.com/67168828/122692653-c9157800-d20c-11eb-8730-ea761c44c113.png)

**Comment by scripting** [comment]
What’s the url of your new site?

**Comment by brandon-toner** [comment]
http://brandontoner.ca

**Comment by scripting** [comment]
Is that pointing to your glitch site?

**Comment by brandon-toner** [comment]
Yup, should be.

I have some placeholder HTML there now while I figure out how to connect it to Little Outliner.

Brandon Toner
***@***.***
(902) 371-0271

**Comment by brandon-toner** [comment]
Yup, should be. I have some placeholder HTML there now while I figure out how to connect it to Little Outliner.

**Comment by scripting** [comment]
Good. I’ll help figure that out in the morning. 😀

**Comment by scripting** [comment]
Good morning Brandon! ;-)

I have the next step for you and it's very easy, and hopefully it works.

1. At the top level of the sub-folder of the domains folder for brandontoner.ca, create a file called config.json.
2. In that file put exactly the text you see in [this file](https://gist.github.com/scripting/bc94224619c9bbe6c608aec1a41222a1). 
3. In your browser, go to http://brandontoner.ca. you should see a picture of a driveway.
4. If that worked, change the address of the driveway picture to the public address of the OPML file containing the content you want to appear at brandontoner.ca.

That's it, you should be ready to go. 

If you want to read about other things you can do in config.json, check out the [configuring page](https://github.com/scripting/pagePark/blob/master/docs/config.md) on the PagePark site. 

Please let me know how it goes.

**Comment by brandon-toner** [comment]
Great! Added the config.json file and successfully displayed the driveway.

I'm back on that work network where Little Outliner is being blocked, so I'll try to add the public link once I'm on another network.

I suspect its the url that displays when I click "Get Public Link".

**Comment by scripting** [comment]
>>I suspect its the url that displays when I click "Get Public Link".

No that's something else. 

1. Choose View OPML in the File menu.
2. The address you're looking for is in the address bar of the browser.

**Comment by brandon-toner** [comment]
Ah! Glad I clarified. Thanks.

**Comment by brandon-toner** [comment]
Ok, did that. It's displaying the raw OPML file when I load http://brandontoner.ca.

How do I render it?

**Comment by scripting** [comment]
Brandon, we are in uncharted territory, I'm going to have to look into this might require a mode to PagePark. I'll let you know.

**Comment by brandon-toner** [comment]
I like the sounds of uncharted territory. I'm guessing the PagePark can render an OPML file, but is not recognizing the url-based OPML generated by Little Outliner for live-syncing?

**Comment by brandon-toner** [comment]
Some updates:
1. I've uploaded the OPML manually (copy-paste into an index.opml file so that the site renders something in the meantime)
2. I've also figured out how to upload an image to the outline by inspecting the XML on one of your pages.
3. I played around with the "collapse" attribute to set some intentional closures on the OPML file. 

I'm guessing I did a few of these the "hard way":
1. Is there an "export" in Little Outliner to generate an OPML?
2. What is the best way to add an image to an outline?

Here's how it looks now:
![image](https://user-images.githubusercontent.com/67168828/122824494-1b12d800-d2b7-11eb-9e46-ccaec021c896.png)

**Comment by brandon-toner** [comment]
An eventual goal would be to have the "Blog" hosted on this domain as well, with something similar to the Scripting News layout.

Key elements:
- Blog
- Links Blog
- About

**Comment by scripting** [comment]
FYI -- I do my linkblog with radio3.io.

**Comment by scripting** [comment]
there's a new version of pagepark that has the mirrors problem fixed.

replace your copy of pagepark.js with the new one.

https://github.com/scripting/pagePark/blob/master/pagepark.js

that's the only part in this release that changed.

you may have to puzzle it out because github and glitch are both new for you.

**Comment by brandon-toner** [comment]
I now have the new version uploaded to Glitch. Getting an error on line 40.
![image](https://user-images.githubusercontent.com/67168828/122985782-f67d3580-d374-11eb-996a-663b8fa63f1e.png)

**Comment by scripting** [comment]
that sucks. i can probably fix that pretty quickly.

**Comment by scripting** [comment]
OK there's a new version of pagepark.js with the "package" error fixed.

https://github.com/scripting/pagePark/blob/master/pagepark.js

It's odd, because I'm pretty sure that was in the previous version too.

Let me know how it goes.

**Comment by brandon-toner** [comment]
line40 error is cleared, getting some on 655-666 (didn't notice whether they were present on the .18 version)

![image](https://user-images.githubusercontent.com/67168828/122988934-a7390400-d378-11eb-8eb3-e5bc3258b848.png)

**Comment by scripting** [comment]
There's a new pagepark.js in the repo.

While you're trying that I'll try running it on glitch.

**Comment by scripting** [comment]
I'm trying to run it on glitch and it's still getting errors, very weird
ones.

I thought I had it running here, not sure what's going on.

**Comment by scripting** [comment]
I got it to run sort of -- I had to change the package.json file.

At the very end, under "engines" for node, I changed it from * to 14.x.

"engines": {
		"node": "14.x"
	}

Then it ran without errors, but the home page didn't show up properly.

Something very weird is going on, I'll try to figure out what it's going to take to get pagepark to run here again properly.

**Comment by scripting** [comment]
Brandon, I gathered my thoughts on Glitch this morning..

http://scripting.com/2021/06/23/145559.html?title=revisitingGlitchAYearLater

You can skip to the end, but I think the rest of it is worth reading.
Digital Ocean may be a better long-term option, and their docs are
excellent, and their systems are standard and reliable.

>

**Comment by scripting** [comment]
Brandon, I just spent another 1/2 hour tinkering with PP on Glitch, and it just doesn't work reliably, and I don't have the tools (or time) I need to figure out why.

**Comment by brandon-toner** [comment]
Hmm — good to know! I'll start looking into Digital Ocean and other serving options.

**Comment by brandon-toner** [comment]
Someone had suggested checking out [GitHub Pages](https://pages.github.com/) as an option for hosting — I also plan to look into this a bit.

**Comment by scripting** [comment]
How would you get the LO2 outlines onto GitHub?

**Comment by brandon-toner** [comment]
Mirroring is not an option in that case? (As we were doing on Glitch)

**Comment by scripting** [comment]
When you use GitHub Pages you're using their web server, not PagePark.

Their server doesn't have the mirrors feature, it certainly can't render
OPML in any nice way.

I have asked them to do it, btw. Eventually I'm sure they will. But not yet.

**Comment by brandon-toner** [comment]
Ah, unfortunate. So that likely narrows down to Digital Ocean as the next step. A worthwhile expense — I'll go for it.

**Comment by brandon-toner** [comment]
I've created an account on Digital Ocean — I'm finding that platform a bit more challenging to get oriented with.

Steps completed: 
- Account created on Digital Ocean
- Domain pointed to Digital Ocean

I took a pass through some of their docs — but couldn't quite get off the ground with setting up the node.js server like on Glitch.

**Comment by scripting** [comment]
Brandon, why don't I just set up your domain here, using a mirror, until we
get to a place where it's easy for you to do your own.

I think we're wasting a bunch of time, both of us, when there's a much
easier path available.

**Comment by brandon-toner** [comment]
> Brandon, why don't I just set up your domain here, using a mirror, until we get to a place where it's easy for you to do your own. I think we're wasting a bunch of time, both of us, when there's a much easier path available.

Yup that sounds like a good plan. That would definitely help me to "get up and running" and "into the flow of it" without having to learn to much code/programming on the front-end. 

Let's do it.

**Comment by scripting** [comment]
good, i'll send you a DM on twitter.
